siphon/catalog.py

```diff
@@ -24,6 +26,106 @@
 log.setLevel(logging.ERROR)
 
 
+class IndexableMapping(OrderedDict):
+    """Extend ``OrderedDict`` to allow index-based access to values."""
+    def __getitem__(self, item):
+        """Return an item either by index or name."""
+        try:
+            item + ''  # Raises if item not a string
```

Comment:
That's pretty much it. And I don't like having to figure out what the right type is (though here it's probably `basestring`).